jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ErrorPageTest.java

```diff
@@ -198,6 +246,56 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
         }
     }
 
+    public static class AsyncSendErrorServlet extends HttpServlet implements Servlet
+    {
+        @Override
+        protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException
+        {
+            try
+            {
+                final CountDownLatch hold = new CountDownLatch(1);
+                final boolean lateComplete = "true".equals(request.getParameter("latecomplete"));
+                AsyncContext async = request.startAsync();
+                async.start(() ->
+                {
+                    try
+                    {
+                        response.sendError(599);
+                        if (!lateComplete)
+                            async.complete();
+                        hold.countDown();
+                        if (lateComplete)
+                        {
+                            // Wait until request is recycled
+                            while (Request.getBaseRequest(request).getMetaData()!=null)
+                            {
+                                try
+                                {
+                                    System.err.println("waiting "+request);
+                                    Thread.sleep(100);
+                                }
+                                catch (InterruptedException e)
+                                {
+                                    e.printStackTrace();
+                                }
+                            }
+                            async.complete();
```

Comment:
@sbordet I separated the complete into the two cases.  This one now throws an ISE because the request/response has already been recycled!  Not sure how to fix? or even if we can?